gbpservice/nfp/base_mode_vm/api/v1/controllers/controller.py

```diff
@@ -107,25 +94,30 @@ def post(self, **body):
 
             # Assuming config list will have only one element
             config_data = body['config'][0]
-            context = config_data['kwargs']['context']
-            request_info = config_data['kwargs']['request_info']
-
-            if config_data['resource'] == 'ansible':
-                nfd = request_info['network_function_data']
-                nf_data = nfd['network_function_details']
-                service_config = nf_data['network_function']['service_config']
+            service_type = body['info']['service_type']
+            if body['config'][0]['resource'] == 'interfaces':
+                generic_config_data = body['config']
+                routes_info = generic_config_data[1]	# routes
```

Comment:
This should not be done in the conditional check for interfaces. Things should work even if the routes comes are first API and followed by interfaces. Split this condition and handle both interfaces and routes.